File: 0048-rotate-image/0048-rotate-image.c
void rotate(int** matrix, int matrixSize, int* matrixColSize) {
    int left = 0, right = matrixSize - 1;

    while (left < right) {
        int top = left, bottom = right;

        for (int i = 0; i < right - left; i++) {
            int temp = matrix[top][left + i];

            matrix[top][left + i] = matrix[bottom - i][left];
            matrix[bottom - i][left] = matrix[bottom][right - i];
            matrix[bottom][right - i] = matrix[top + i][right];
            matrix[top + i][right] = temp;
        }

        left++;
        right--;
    }
}

File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int n=matrix.length;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                int temp=matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
        for(int row=0;row<n;row++){
            int startcol=0;
            int endcol=n-1;
            while(startcol<=endcol){
                int temp=matrix[row][startcol];
                matrix[row][startcol]=matrix[row][endcol];
                matrix[row][endcol]=temp;
                startcol++;
                endcol--;
            }
        }
    }
}